### PR TITLE
[TASK] Add progressbar for all FileProcessors

### PR DESCRIPTION
**Pull request by sabbelasichon**
Resolves: #6380

### CODE REVIEW

**Comment by sabbelasichon** [comment]
We have a chicken egg problem here. Because the NeonFileProcessor must implement the new contract. What do you think about it, besides the fact that every test fails?

File: rules/Composer/Application/FileProcessor/ComposerFileProcessor.php [hunk 1]
```diff
@@ -45,14 +45,14 @@ public function __construct(
     /**
      * @param File[] $files
      */
-    public function process(array $files): void
+    public function process(array $files, callable $advanceProgressBar): void
```
Comment (TomasVotruba) [inline rules/Composer/Application/FileProcessor/ComposerFileProcessor.php:48]: `callable` is very poor contract that would shoot you back with confusing error message. In general, we should code against interface.

**Comment by sabbelasichon** [comment]
I am on it. I think i will come up with a good and clear solution after i revisited the current processors again

**Comment by sabbelasichon** [comment]
Closing this in favour of a new try

**Pull request closed without merging**